### PR TITLE
docs: Split the YAML reference in multiple sections

### PR DESCRIPTION
**Pull request by mnapoli**
This PR splits this documentation page into multiple sections and code blocks: https://www.serverless.com/framework/docs/providers/aws/guide/serverless.yml

List of all changes:

- Split the big code block in sub-blocks and sections
- Had to split large sections, like `provider` and Lambda events into sub-section else we're back to a big code block
- Moved all comments above the line, instead of "in-line", because it added a horizontal scrollbar
- Harmonized (as much as possible) details like quotes, dots, comment formatting, showing default values…
- Improved wording of most comments
- Re-ordered some options so that the most common ones are visible first

I did not change the content of the examples except for the details above. Some sections could be rewritten almost entirely, but that's not the point here.

Also eventually we might want to move these references into each related documentation (e.g. move the "HTTP API" config reference into the HTTP API doc page), but that's also another topic.

### CODE REVIEW

**Comment by pgrzesik** [comment]
Hello @mnapoli 👋 is this one ready for a review? There seems to be a small conflict with current `master` branch

**Comment by mnapoli** [comment]
TODO:

- [ ] Resolve conflict
- [ ] Add small intro to each section
- [ ] Final review

File: docs/providers/aws/guide/serverless.yml.md
Comment (mnapoli) [inline docs/providers/aws/guide/serverless.yml.md:64]: I want to make this change:

```suggestion
  # The AWS profile to use to deploy (default: "default" profile)
```

Could anyone confirm this is correct?

File: docs/providers/aws/guide/serverless.yml.md
Comment (mnapoli) [inline docs/providers/aws/guide/serverless.yml.md:122]: Can I get a confirmation on the "default: forever"?

File: docs/providers/aws/guide/serverless.yml.md
Comment (mnapoli) [inline docs/providers/aws/guide/serverless.yml.md:128]:
```suggestion
  # Use function versioning (default: false)
```

is that correct?

**Comment by mnapoli** [comment]
I rebased and changed the target branch to v3.

**Comment by mnapoli** [comment]
I get very weird failures in GitHub Actions:

```
Run npm run prettier-check:updated

> serverless@3.0.0-pre prettier-check:updated
> pipe-git-updated --base=v3 --ext=css --ext=html --ext=js --ext=json --ext=md --ext=yaml --ext=yml -- prettier -c

'v3' is not in the working tree
Error: Process completed with exit code 128.
```

and

```
 Error: Cannot find module '@serverless/utils/log'
```

Any idea?

**Comment by pgrzesik** [comment]
@mnapoli Did you change the base branch after first having this PR for `master? I think the safest would be to close this PR and open a fresh one against `v3` directly - there also seem to be a small conflict

**Comment by mnapoli** [comment]
@pgrzesik no need, I just rebased again against v3. I think this was caused by `v3` being rebased again in the meantime.

File: docs/providers/aws/guide/serverless.yml.md
Comment (pgrzesik) [reply docs/providers/aws/guide/serverless.yml.md:77]: yes, by default we're using `default` AWS profile

File: docs/providers/aws/guide/serverless.yml.md
Comment (pgrzesik) [inline docs/providers/aws/guide/serverless.yml.md:89]: That is a bit misleading - we only apply these tags to API GW and functions and nothing else

File: docs/providers/aws/guide/serverless.yml.md
Comment (pgrzesik) [reply docs/providers/aws/guide/serverless.yml.md:135]: the default is the CloudFormation/AWS default - we don't set any specific value if it's not provided, so based on CW docs it's "forever" https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/WhatIsCloudWatchLogs.html

File: docs/providers/aws/guide/serverless.yml.md
Comment (pgrzesik) [reply docs/providers/aws/guide/serverless.yml.md:141]: by default we version functions

**Review by pgrzesik — COMMENTED** [review]
I have a few comments but in general it looks good 👍

File: docs/providers/aws/guide/serverless.yml.md
Comment (mnapoli) [reply docs/providers/aws/guide/serverless.yml.md]: Thanks, I clarified that

**Comment by mnapoli** [comment]
Thanks @pgrzesik I applied all your suggestions

**Review by pgrzesik — APPROVED** [review]
Looks good to me 👍 Could be also good to have a review from @medikoo before merging

**Review by medikoo — COMMENTED** [review]
@mnapoli that's a great improvement! I've proposed few minor updates

File: docs/providers/aws/guide/serverless.yml.md
Comment (mnapoli) [reply docs/providers/aws/guide/serverless.yml.md:27]: This could be useful especially with v3 when you know your service is compatible with v2 and v3, yet you don't want to force everyone to use either v2 or v3 (e.g. in a team where devs have different versions installed).

File: docs/providers/aws/guide/serverless.yml.md
Comment (mnapoli) [reply docs/providers/aws/guide/serverless.yml.md:32]: Good point, yes this is intentional: since we now always document the default value, it makes less sense to duplicate the default value below.

On top of that, the main use case where you want to copy a line from the docs and paste it into your project is to override the default. E.g. on this line here the main use case is for users to opt-in the error mode. So it makes more sense to document a different value (when possible, the most popular alternative to the default).

File: docs/providers/aws/guide/serverless.yml.md
Comment (mnapoli) [reply docs/providers/aws/guide/serverless.yml.md:34]: OK I will rebase (it would have been easier to remove it in this PR because of the git conflicts).

File: docs/providers/aws/guide/serverless.yml.md
Comment (medikoo) [reply docs/providers/aws/guide/serverless.yml.md:27]: > This could be useful especially with v3 when you know your service is compatible with v2 and v3

That is best expressed with `2 || 3` (so 4, 5 etc are not recognized as supported)

File: docs/providers/aws/guide/serverless.yml.md
Comment (medikoo) [reply docs/providers/aws/guide/serverless.yml.md]: In v3 branch (as in every PR) for a good later historical overview, we do our best to keep all changes that relate to specific functionality in the single specific commit.

e.g. above removal was added to commit which upgraded local fallback.

Having that we can clearly see all changes that were made that relate to a given functionality

File: docs/providers/aws/guide/serverless.yml.md
Comment (mnapoli) [reply docs/providers/aws/guide/serverless.yml.md]: Thanks, that's moved!

File: docs/providers/aws/guide/serverless.yml.md
Comment (mnapoli) [reply docs/providers/aws/guide/serverless.yml.md]: Same, moved!

**Review by medikoo — COMMENTED** [review]
@mnapoli sorry, I was integrating latest master changes into v3 and applying some cleanup to commit, so it needs another rebase, but I won't touch it now until this one is merged :)

**Comment by mnapoli** [comment]
No problem, I just rebased.

File: docs/providers/aws/guide/serverless.yml.md
Comment (mnapoli) [reply docs/providers/aws/guide/serverless.yml.md:27]: Right, I understand that `^2.67 || 3` is technically better, but it's much simpler and obvious for users to go with `>=2.67`, and it's not that risky since major versions are so far apart in time. The thing is that not everyone is familiar with semver range constraints.

Comment (medikoo) [reply docs/providers/aws/guide/serverless.yml.md:27]: > it's not that risky since major versions are so far apart in time.

It's risky, and just that is a valid concern imo. It's unlikely for users to revisit the range, then, while many services are destined to be in service for years simply in maintenance mode.

For same reason, you never put such a range in packages published on npm. We're giving poor's man advice that way.

Also I don't understand the argument that `^2.67 || 3` is more "hard" to use. It looks very clean and straightforward to me. 

>  The thing is that not everyone is familiar with semver range constraints.

It's exactly why, we should provide a future-proof examples, that will prevent users running into issues, long term.

**Review by medikoo — APPROVED** [review]
Great improvement 👍

**Pull request merged**
